### PR TITLE
Configurable browse-by

### LINKED ISSUE #361: Configurable Browse-By links
The Browse-By links added by #360 are currently hardcoded. 
There should be a way to add these through config, defining which metadata definitions to display a link to browse-by for.

### PR DESCRIPTION
**Pull request by Atmire-Kristof**
This PR closes #361

Previously, the browse-by links in the navigation menu and community/collection pages were hard-coded. This PR adds config to dynamically load these links.

### Config
The config for the browse-by links is defined at `browseBy.types` in the form of an array. Each link is resembled by an object containing 3 possible properties.

"metadata": The browse id to use for fetching info from the rest api (e.g. title, dateissued, author…)
"type" : The type of Browse-By page to display (currently 3 types to choose from: title, date and metadata)
"metadataField" : The metadata-field used to create starts-with options (only necessary when the type is set to 'date')
Current config for the links:

```
    types: [
      {
        metadata: 'title',
        type: 'title'
      },
      {
        metadata: 'dateissued',
        type: 'date',
        metadataField: 'dc.date.issued'
      },
      {
        metadata: 'author',
        type: 'metadata'
      },
      {
        metadata: 'subject',
        type: 'metadata'
      }
    ]
```

### CODE REVIEW

**Comment by paulo-graca** [comment]
Like @tdonohue, I didn't found any issue with this PR and the overall code looks to me.
The only question I would like to make is why did you choose 'metadata' for the type ("The type of Browse-By page to display") of stuff that is presented in a categorized way. If we look to our legacy version, the predefined types are related with the fields themselves. I'm not suggesting this approach. I'm just saying that the name 'metadata', to me, is not much suggestive for what it will be showed. I think "facet" or "category" will be a most suited name to replace it. 
This is my suggestion:
```
      {
        metadata: 'subject',
        type: 'category'
      } 
```
But I don't want to block this PR. I just want to hear from you @Atmire-Kristof if there were a special reason to call it 'metadata'.

**Comment by artlowel** [comment]
@paulo-graca it's called `metadata` because the rest response has a boolean `metadataBrowse`, if that is true, the type will be `metadata`. 

```json
      {
        "id": "subject",
        "metadataBrowse": true,
```

So if we want to change it, I suggest we change it both in the rest and the UI at the same time.

I do concede that it's confusing having a property in the config called `metadata` and then having `metadata` also be a value for the `type` property. So I asked @Atmire-Kristof to change the property name to `id` to correspond with the rest response as well.

File: config/environment.default.js [hunk 1]
```diff
@@ -149,7 +149,32 @@ module.exports = {
...
+    defaultLowerLimit: 1900,
+    // List of all the active Browse-By types
+    // Adding a type will activate their Browse-By page and add them to the global navigation menu, as well as community and collection pages
+    // Allowed fields and their purpose:
+    //    id:             The browse id to use for fetching info from the rest api
+    //    type:           The type of Browse-By page to display
+    //    metadataField:  The metadata-field used to create starts-with options (only necessary when the type is set to 'date')
+    types: [
+      {
+        id: 'title',
```
Comment (paulo-graca) [inline config/environment.default.js]: I think it helps this change.

File: config/environment.default.js [hunk 2]
```diff
@@ -149,7 +149,32 @@ module.exports = {
...
+        type: 'title'
+      },
+      {
+        id: 'dateissued',
+        type: 'date',
+        metadataField: 'dc.date.issued'
+      },
+      {
+        id: 'author',
+        type: 'metadata'
```
Comment (paulo-graca) [inline config/environment.default.js]: I showed this to others, and also they were a bit confused about what it is `type: 'metadata'`. But now I understand the reason you choose metadata. I think we could live with it :)

**Comment by tdonohue** [comment]
This is at +3, merging

**Pull request merged**